applications/StructuralMechanicsApplication/custom_elements/cr_beam_element_3D2N.cpp

```diff
@@ -368,34 +364,31 @@ namespace Kratos
 		Kd(5, 5) = 3.0 * E * Iz * Psi_z / L;
 
 
-		//add geometric stiffness part
-		if (this->mIsLinearElement == false)
-		{
-			const double l = this->CalculateCurrentLength();
-			const double N = this->mNodalForces[6];
+		const double l = this->CalculateCurrentLength();

```

Comment:
Here the method is still called CalculateCurrentLenght(), whic is the difference?, besides, maybe could be a good idea to have a common base class for all beams and trusses (idem for shells)

Reply:
base class for shells is coming soon, working on it

Reply:
I also thought about a base class for 1d elements but I decided not to do it due to too many differences.